Theme 6 hw

GrimmFndg opened the pull request:
Не смог выполнить Домашнюю работу. Никак не удалось реализовать консоль на сервере.
Не совсем понял как работают сокеты, и в целом работа с сетью.Пытался реализовать метод через Scanner, но ничего кроме ошибок не получал. Пробовал со своим интерфейсом, с кодом разобранном на уроке, и кодом из методички. Буду дальше разбираться с сокетами. Если возможно - сдам работу на проверку позднее.

GrimmFndg@gmail.com

GrimmFndg committed:
GrimmFndg@gmail.com
Выполнил все пункты, кроме преобразования String в Int

GrimmFndg committed:
GrimmFndg@gmail.com

GrimmFndg committed:
Выполнил все пункты.
GrimmFndg@gmail.com

GrimmFndg committed:
Git

GrimmFndg committed:
Clear junk

GrimmFndg committed:
Git

GrimmFndg committed:
Реализована очистка текстового поля после ввода сообщения.

GrimmFndg@gmail.com

GrimmFndg committed:
GrimmFndg@gmail.com

GrimmFndg committed:
GrimmFndg@gmail.com

GrimmFndg committed:
GrimmFndg@gmail.com

GrimmFndg committed:
GrimmFndg@gmail.com

GrimmFndg committed:
GrimmFNdg@gmail.com

GrimmFndg committed:
GrimmFndg@gmail.com

usharik commented on Theme_6/ServerWindow.java:
```diff
@@ -0,0 +1,94 @@
+                try {
+                    out.println("end");
+                    out.flush();
+                    sock.close();
```
Вы сначала закрыли сокет, а потом полученные из него потоки. В чем смысл так делать?

usharik commented on Theme_6/ServerWindow.java:
```diff
@@ -0,0 +1,94 @@
+                    sock.close();
+                    out.close();
+                    in.close();
+                } catch (IOException exc) {
```
Пустой блок catch - очень большое зло. Обязательно выводите текст ошибки через ex.printStackTrace();

usharik commented on Theme_6/ServerWindow.java:
```diff
@@ -0,0 +1,94 @@
+                        jta.append("\n");
+                    }
+                }
+            } catch (Exception e) {
```
Не делайте пустых блоков catch

GrimmFndg committed:
Проведена работа над ошибками.
Программа корректно работает и выполняет требуемую задачу.

Проведен рефакторинг кода разобранного на уроке с целью улучшения читаемости.
Добавлен функционал отслеживания текущего количества клиентов в статусе "Онлайн":
Консоль сообщает о количестве пользователей при подключении клиента.
Консоль сообщает о количестве пользователей после отключения клиента.

GrimmFndg@gmail.com

GrimmFndg committed:
Удалены лишние файлы
